### ISSUE TITLE
S3TransferManager ability to set endpointOverride

### ISSUE THREAD

**Issue by ehmurray8** [aws/aws-sdk-java-v2#2740]
## Describe the Feature
With the S3AsyncClient I am able to set the endpointOverride like this.
```
S3AsyncClient.builder().endpointOverride(URI("http://minio-s3:9000"))
```
I can't find a way to do the same thing with S3TransferManager.

## Is your Feature Request related to a problem?
When running my project locally I am using Minio to mock S3 locally, and I need to be able to send requests to it.

## Proposed Solution
I would like to be able to do the same thing with S3TransferManager
```
 S3TransferManager.builder()
        .s3ClientConfiguration { cfg ->
            cfg.endpointOverride(URI("http://minio-s3:9000"))
        }
```

## Your Environment
* AWS Java SDK version used: 2.17.46-PREVIEW
* JDK version used: 16.0.2
* Operating System and version: Mac M1

**Comment by debora-ito** [comment]
Thank you for reaching out @ehmurray8. Feature request acknowledged.

**Comment by aceeric** [comment]
This would be valuable for us as well. Presently, we are using the `S3AsyncClient` to download 10K-sized objects from Minio because the class supports the endpoint override. The best we can get this way is 4MiB/sec for downloads. In comparison, using the [s5cmd](https://github.com/peak/s5cmd) utility as a performance benchmark, we are able to get 38MiB/sec performing the same downloads. I'm wondering if the difference is partially due to the S3 Transfer Manager's optimization of multipart downloads.

**Comment by openwebsolns** [comment]
Is there an update on this feature request?

**Comment by zoewangg** [comment]
Implemented in #3084. Closing the issue
